src/main/java/org/veupathdb/service/eda/compute/plugins/correlation/CorrelationPlugin.java

```diff
@@ -0,0 +1,332 @@
+        VariableDef revisedComputeEntityIdVarSpec = isSameEntity ? computeEntityIdVarSpec : parentEntityIdVarSpec;
+        String revisedComputeEntityIdColName = util.toColNameOrEmpty(revisedComputeEntityIdVarSpec);
+        CollectionDef collection2 = metadata.getCollection(assay2).orElseThrow();
+
+        List<VariableDef> entity2IdColumns = new ArrayList<>();
+        for (EntityDef ancestor : metadata.getAncestors(entity2)) {
+          entity2IdColumns.add(ancestor.getIdColumnDef());
+        }
+        // if were not on the same entity, we can remove the assay entity id from the id columns
+        if (!isSameEntity) {
+          entityIdColumns.remove(revisedComputeEntityIdVarSpec);
+          entity2IdColumns.remove(entity2IdVarSpec);
+        }
+
+        connection.voidEval("assayData <- " + INPUT_DATA);
+        List<VariableSpec> assay2InputVars = ListBuilder.asList(revisedComputeEntityIdVarSpec);
+        assay2InputVars.addAll(util.getCollectionMembers(assay2));
+        assay2InputVars.addAll(entity2IdColumns);
+        connection.voidEval(util.getVoidEvalFreadCommand(INPUT_2_DATA, assay2InputVars));
+        connection.voidEval("assay2Data <- " + INPUT_2_DATA);
+
+        List<String> dotNotatedEntity2IdColumns = entity2IdColumns.stream().map(VariableDef::toDotNotation).toList();
+        String dotNotatedEntity2IdColumnsString = util.listToRVector(dotNotatedEntity2IdColumns);
+
+        String collection2MemberType = collection2.getMember() == null ? "unknown" : collection2.getMember();
+        // If either collection is an eigengene, we'll use our base correlation function in veupathUtils,
+        // so we want to set the isEigengene flag to true.
+        if (collection2MemberType.toLowerCase().contains("eigengene")) {
+          isEigengene = true;
+        }
+
+        // Prep data and run correlation
+        if (isEigengene) {
+          // If we have eigenegene data, we'll use our base correlation function in veupathUtils, so we
+          // only need to make data frames for the assay data and sample metadata.
+          connection.voidEval("data1 <- assayData; " + 
+            "data1 <- data1[order(" + revisedComputeEntityIdColName + ")]; " + 
+            "data1 <- data1[, -as.character(" + dotNotatedEntityIdColumnsString + "), with=FALSE];" +
+            "data1 <- data1[, -" + singleQuote(revisedComputeEntityIdColName) + ", with=FALSE]");
+
+          connection.voidEval("data2 <- assay2Data; " +
+            "data2 <- data2[order(" + revisedComputeEntityIdColName + ")]; " + 
+            "data2 <- data2[, -as.character(" + dotNotatedEntity2IdColumnsString + "), with=FALSE];" +
+            "data2 <- data2[, -" + singleQuote(revisedComputeEntityIdColName) + ", with=FALSE]");
+
+          connection.voidEval("computeResult <- veupathUtils::correlation(data1=data1, data2=data2" +
+                                                              ", method=" + singleQuote(method) +
```

Comment:
i dont remember what you said about prefilters - are you planning to fix them in a different PR? Totally fine either way just curious!

Reply:
I think once those are in we can move the correlation calculation outside the `isEigengene` conditional because they should be the same, no?

Reply:
thats a separate pr, coming soon hopefully. i wanted it to be this week, but that doesnt look like its going to happen. and yes, i think if i manage to do it well enough, we can build the data objects in the conditional and do the correlation calculation outside it.